airflow/models.py

```diff
@@ -1288,19 +1288,19 @@ def run(
             verbose=True)
 
         if not runnable and not mark_success:
-            if self.state != State.QUEUED:
-                # If a task's dependencies are met but it can't be run yet then queue it
-                # instead
-                self.state = State.QUEUED
-                msg = "Queuing attempt {attempt} of {total}".format(
-                    attempt=self.try_number % (task.retries + 1) + 1,
-                    total=task.retries + 1)
-                logging.info(hr + msg + hr)
-
-                self.queued_dttm = datetime.now()
-                msg = "Queuing into pool {}".format(self.pool)
-                logging.info(msg)
-                session.merge(self)
+            # we might have hit concurrency limits, which means we probably
+            # have been running prematurely.
+            self.state = State.UP_FOR_RETRY
```

Comment:
@aoen Trying to get out the (seemingly) standoff. Any reasons why we don't include "RUN_DEPS" at scheduling time (maybe to be added in _execute_task_instances)? We do so for backfills.

Otherwise I will go for NONE as a state, as this is the input state for the scheduler together with UP_FOR_RETRY. And mark it to do.